ObjectiveCloudant/Operations/CDTQueryViewOperation.h

```diff
@@ -0,0 +1,193 @@
+    /**
+     Allow stale views.
+     **/
+    CDTStaleViewOk,
+    /**
+     Allow stale views, but update them immediately after the request.
+     **/
+    CDTStaleViewUpdateAfter
+};
+
+/**
+ An operation to query Map Reduce views.
+ */
+@interface CDTQueryViewOperation : CDTCouchDatabaseOperation
+
+/**
+ * The name of the design document which contains the view.
+ *
+ * Required: This needs to be set before an operation can successfully run.
+ */
+@property (nullable, nonatomic, strong) NSString *ddoc;
+
+/**
+ * The name of the view to query.
+ *
+ * Required: This needs to be set before an operation can succesfully run.
+ */
+@property (nullable, nonatomic, strong) NSString *viewName;
+
+/**
+ * Return the documents in 'descending by key' order.
+ *
+ * Default : NO, order is ascending by key.
+ */
+@property (nonatomic) BOOL descending;
+
+/**
+ * Stop the view returning results when the specified key is reached.
+ * It must be set to an Object of type `NSString*` or `NSArray<NSString*>*`.
+ *
+ * Optional: CouchDB will return all the results if this is not set.
+ */
+@property (nullable, nonatomic, strong) NSObject *endKey;
+
+/**
+ * Stop the view returning results when the specified document ID is reached.
+ *
+ * Optional: CouchDB will return all the results if this is not set.
+ */
+@property (nullable, nonatomic, strong) NSString *endkeyDocId;
+
+/**
+ * Group the results of a reduce based on their keys.
+ *
+ * Default: NO.
+ */
+@property (nonatomic) BOOL group;
+
+/**
+ * Group reduce results for documents with complex keys. Using the first x number of items in the
+ * complex key array, where x is the group level.
+ *
+ * For example, if you had the complex key : ["A","B", "C", "D"], if groupLevel is set to 3,
+ * "A", "B" and "C" would be used to group the view results.
+ *
+ * Optional: CouchDB will use the default groupLevel if grouping is enabled, negative values will
+ * result in the parameter not being included in requests
+ **/
+@property (nonatomic) NSInteger groupLevel;
+
+/**
+ * Include the full document content in the view results.
+ *
+ * Default: NO
+ */
+@property (nonatomic) BOOL includeDocs;
+
+/**
+ * Include rows with the specified `endKey`.
+ *
+ * Default: NO.
+ */
+@property (nonatomic) BOOL inclusiveEnd;
+
+/**
+ * Return only documents that match the specified key.
+ * The key must be of type `NSString*` or `NSArray<NSString*>*`
+ *
+ * Cannot be used with `keys` option.
+ *
+ * Optional: CouchDB will return all documents emitted from the view.
+ */
+@property (nullable, nonatomic, strong) NSObject *key;
+
+/**
+ * Return only the documents that match the specified keys.
+ * The keys in the array must be of type `NSString*` or `NSArray<NSString*>*`
+ *
+ * Cannot be used with `key` option.
+ *
+ * Optional: CouchDB will return all documents emitted from the view.
+ */
+@property (nullable, nonatomic, strong) NSArray<NSObject *> *keys;
+
+/**
+ * Limit the number of documents returned from the view.
```

Comment:
You need to rewrite a bunch of these doc comments to refer to rows in the index rather than documents. I think mostly its just a find/replace type of thing, but double-check the accuracy of the docs.